### ISSUE TITLE
/start requests never time out

### ISSUE THREAD

**Issue by halter73** [SignalR/SignalR#3118]
### Functional impact
- If [for some reason](https://github.com/SignalR/SignalR/issues/3116) a /start request does not complete, the client will never time out the request. This causes the client to remain perpetually in the connecting state.
### Minimal repro steps
- Checkout out the branch for SignalR 2.1.1 (currently release)
- Create a PersistentConnection or Hub with an `OnConnected` method that never completes.
- Try to connect to the PersistentConnection or Hub with the JS or .NET client.
### Expected result
- The /start request should be cancelled after some predefined (configurable?) timeout, and the SignalR client should move from the connecting to the disconnected state.
### Actual result
- The SignalR client hangs in the connecting state indefinitely waiting on the /start request to complete.

**Comment by Joe-U-Questionmark** [comment]
Did this happen? I see it was considered for V3 and then closed without having a changeset attached.. 
We found the same issue recently that when our users use certain secure proxies the /start request never returns and SignalR effectively freezes.

**Comment by Joe-U-Questionmark** [comment]
@halter73 Did this happen? I see it was considered for V3 and then closed without having a changeset attached.. 
We found the same issue recently when our users use certain secure proxies the /start request never returns and SignalR effectively freezes. All requests should have a timeout.

**Comment by halter73** [comment]
@Joe-U-Questionmark This hasn't been fixed.

**Comment by Joe-U-Questionmark** [comment]
@halter73 , Is there a plan to fix it? 
 
We have disabled websockets in production as we had reports from users from large companies with secure proxies that after making a few websocket connections (as they navigated from page to page on a SignalR enabled site) that the Websocket connection and every web request in general were failing (disabling websockets solved the problem)
 
The actual issue was:
*The last message our users got in the console was "SignalR: webSockets transport connected. Initiating start request."
*The http ajax request to "/signalr/start?transport=webSockets" from the function  connection._.startRequest (lines 1435 & 1154) does not have a timeout (as AjaxDefaults has timeout set to null on line 66). 
*As all the requests in the proxy have been exhausted the "/signalr/start?transport=webSockets" request never responds and as there is no timeout the page hangs at this point.
 
If SignalR had a sensible timeout for the "/signalr/start?transport=webSockets" request, closed the half open websocket and fell back to the another protocol this would partly resolve the solution.

I plan on raising a separate issue to cover the specific proxy issue when I have more details.

**Comment by Joe-U-Questionmark** [comment]
@halter73 , Is there a plan to fix it? Why close the issue if it is not fixed? Does that mean you dont plan on fixing it?
